src/components/application_manager/src/commands/mobile/add_command_request.cc

```diff
@@ -531,7 +526,31 @@ bool AddCommandRequest::IsWhiteSpaceExist() {
 }
 
 bool AddCommandRequest::BothSend() const {
-  return send_vr_ && send_ui_;
+    return send_vr_ && send_ui_;
+}
+
+std::string MergeInfos(const std::string& first, const std::string& second) {
+  if (!first.empty() && second.empty()) {
+    return first;
+  }
+  if (first.empty() && !second.empty()) {
+    return second;
+  }
+  if (!first.empty() && !second.empty()) {
+    return first + ", " + second;
+  }
+  return std::string();
+}
+
+const std::__cxx11::string AddCommandRequest::GenerateMobileResponseInfo() {
+    if (hmi_apis::Common_Result::UNSUPPORTED_RESOURCE == vr_result_) {
+        return MergeInfos(vr_info_, ui_info_);
+    }
+
+    if (hmi_apis::Common_Result::UNSUPPORTED_RESOURCE == ui_result_) {
```

Comment:
@LuxoftAKutsan, whitch purpose you use this `if`, if on line 553 you make it anyway. I prefer delete ln550-552.